### PR TITLE
Switch over to using ros_control

### PR DESCRIPTION
**Pull request by mdepp**
In preparation for trying to get the arm working, I'm trying to switch over everything to use ros_control. I've subclassed RobotHW in the new spear_hardware package to define a hardware interface for CASE. As of now it can read wheel velocity commands (e.g. from diff_drive_controller) and send them over to can_mapper (ideally it would send commands directly over CAN).

### CODE REVIEW

**Comment by mdepp** [comment]
Quick note -- trying to start a controller in robot.cpp does not work (just freezes), even though it works fine when done using /controller_manager/switch_controller. Starting the controller in a new thread doesn't work either.

**Comment by mdepp** [comment]
Note -- this is not tested. There's no way to test the odometry stuff right now, but I did test an earlier version of the other stuff on the tx2 (before I moved it away from being its own package)

File: spear_rover/src/hardware.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      const auto velocity_handle = JointHandle(state_handle, &wheel_info.cmd);
+      wheel_state_interface.registerHandle(state_handle);
+      wheel_velocity_interface.registerHandle(velocity_handle);
+    }
+    registerInterface(&wheel_state_interface);
+    registerInterface(&wheel_velocity_interface);
+
+    wheel_commands_publisher =
+        nh.advertise<ActuatorArrayCommand>("/drive/cmds", 1);
+    boost::function<void(const WheelOdomPtr &)> odom_callback =
```
Comment (davidlenfesty) [inline spear_rover/src/hardware.cpp:50]: I really need to learn some of the boost libraries... this is nice

**Comment by mdepp** [comment]
I tested an earlier version of the ros_control interface (minus the odometry stuff) on the tx2 and the uavcan GUI to make sure it was sending CAN messages. Then I moved the files around a bit so it's possible I broke something there. The odometry stuff is not tested at all.

Regarding the commits I'll just do a squash and merge.

**Comment by mdepp** [comment]
Tested again on the tx2. Had to fix a typo, but other than that everything works as before (again, didn't test the odometry stuff). I'll merge it in once the build finishes.

**Comment by mdepp** [comment]
Oops I accidentally did a non-squash merge. Looks like undoing it would be nontrivial though.

**Pull request merged**